mappings/net/minecraft/client/option/SimpleOption.mapping

```diff
@@ -25,36 +96,82 @@ CLASS net/minecraft/class_7172 net/minecraft/client/option/SimpleOption
+		COMMENT Creates a boolean option.
 		ARG 0 key
+		ARG 1 tooltipFactoryGetter
 		ARG 2 defaultValue
 	METHOD method_41750 ofBoolean (Ljava/lang/String;Lnet/minecraft/class_7172$class_7307;ZLjava/util/function/Consumer;)Lnet/minecraft/class_7172;
+		COMMENT Creates a boolean option.
 		ARG 0 key
+		ARG 1 tooltipFactoryGetter
 		ARG 2 defaultValue
 		ARG 3 changeCallback
 	METHOD method_41751 ofBoolean (Ljava/lang/String;ZLjava/util/function/Consumer;)Lnet/minecraft/class_7172;
+		COMMENT Creates a boolean option.
 		ARG 0 key
 		ARG 1 defaultValue
 		ARG 2 changeCallback
 	METHOD method_41752 (Ljava/lang/Boolean;)V
 		ARG 0 value
 	METHOD method_41753 getValue ()Ljava/lang/Object;
+		COMMENT {@return the option's current value}
 	METHOD method_41754 getCallbacks ()Lnet/minecraft/class_7172$class_7178;
-	METHOD method_42399 emptyTooltipFactoryGetter ()Lnet/minecraft/class_7172$class_7307;
+	METHOD method_42399 emptyTooltip ()Lnet/minecraft/class_7172$class_7307;
+		COMMENT {@return the getter for the {@code tooltipFactoryGetter} parameter of the constructor
+		COMMENT to indicate empty tooltips}
 	METHOD method_42401 (Lnet/minecraft/class_310;)Lnet/minecraft/class_7172$class_7277;
 		ARG 0 client
 	METHOD method_42402 ofBoolean (Ljava/lang/String;Z)Lnet/minecraft/class_7172;
+		COMMENT Creates a boolean option.
 		ARG 0 key
 		ARG 1 defaultValue
 	METHOD method_42404 getCodec ()Lcom/mojang/serialization/Codec;
 	METHOD method_42405 (Ljava/lang/Boolean;)V
 		ARG 0 value
 	METHOD method_42406 (Ljava/lang/Object;)Ljava/util/List;
 		ARG 0 value
+	METHOD method_42714 wrapLines (Lnet/minecraft/class_310;Lnet/minecraft/class_2561;)Ljava/util/List;
+		ARG 0 client
+		ARG 1 text
+	METHOD method_42715 (Lnet/minecraft/class_7172$class_7303;Ljava/lang/Object;)Lnet/minecraft/class_2561;
+		ARG 2 value
+	METHOD method_42716 (Ljava/util/List;Ljava/lang/Object;)Ljava/util/List;
+		ARG 1 value
+	METHOD method_42717 constantTooltip (Lnet/minecraft/class_2561;)Lnet/minecraft/class_7172$class_7307;
+		COMMENT {@return the getter for the {@code tooltipFactoryGetter} parameter of the constructor
+		COMMENT to indicate constant tooltips}
+		ARG 0 text
+	METHOD method_42718 (Lnet/minecraft/class_2561;Lnet/minecraft/class_7291;)Lnet/minecraft/class_2561;
+		ARG 0 optionText
+		ARG 1 value
+	METHOD method_42719 (Lnet/minecraft/class_2561;Lnet/minecraft/class_310;)Lnet/minecraft/class_7172$class_7277;
+		ARG 1 client
+	METHOD method_42720 enumValueText ()Lnet/minecraft/class_7172$class_7303;
+		COMMENT {@return the getter for the {@code valueTextGetter} parameter of the constructor
+		COMMENT to use the value's text as the option's rendered text}
+		COMMENT
+		COMMENT @apiNote This requires the value to be an enum extending {@link TranslatableOption}.
 	CLASS class_7173 PotentialValuesBasedCallbacks
+		COMMENT A set of callbacks for a cycling option.
+		COMMENT
+		COMMENT @see <a href="#callbacks">Callbacks</a>
```

Comment:
Isn't this link invalid since javadoc generates an html page for each inner class?

Reply:
Hmm, how do we link to that then?

Reply:
Hmm, I can think of two ways.

```suggestion
		COMMENT @see <a href="SimpleOption.html#callbacks">Callbacks</a>
```

You can include the filename to specify the class. Intellij IDEA is smart enough to render this link properly. Eclipse and VSCode don't support the `<a>` tag anyway.

Alternatively, you could move docs to the `Callbacks` class and link to it instead.